### DISCUSSION TITLE
GitHub stats question

### DISCUSSION THREAD

**Discussion by cxhello** [anuraghazra/github-readme-stats#2864] [Q&A] [answered]
**The results shown in my profile are wrong, but the actual link to visit is correct. Is there something wrong with my configuration?**

![image](https://github.com/anuraghazra/github-readme-stats/assets/49056040/4d3b37bc-f12d-4d83-9a70-ddd78cf2f4dd)

![image](https://github.com/anuraghazra/github-readme-stats/assets/49056040/f716a684-9be1-42d2-a846-c20726bfbbc2)

**this is my config**

![image](https://github.com/anuraghazra/github-readme-stats/assets/49056040/842e132b-8108-4b9d-b734-a39f27e2d4e2)

一名普通的 Java 程序员

![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=cxhello&include_all_commits=true&hide_title=true&show_icons=true&theme=material-palenight)
![Top Langs](https://github-readme-stats.vercel.app/api/top-langs/?username=cxhello&layout=compact&theme=material-palenight)

**Comment by rickstaa** [comment, marked as answer]
Two things cause your card to show up with `0` commits. First, the GitHub rest API which we use to fetch the total commits when `include_all_commits=true` sometimes fails (see https://github.com/anuraghazra/github-readme-stats/issues/1515#issuecomment-1256977725). In that case, the code returns `0`. Because we use caching for the cards, this incorrect value stays on the card for 4 hours before being fixed. When you request the card through the URL, you likely use a slightly different syntax, breaking the caching mechanism. You can see this is the case by appending a random string (e. g. `random=2345545345`).

[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=cxhello&include_all_commits=true&random=2345545345)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=cxhello&include_all_commits=true&random=2345545345)](https://github.com/anuraghazra/github-readme-stats)
```

**Reply by cxhello** [reply]
Thank you for your reply. If I want him to always display the correct value, can I only display the number of commits for this year by including_all_commits=false?

**Reply by rickstaa** [reply]
Currently, that is the best way to solve this without creating a PR to mitigate this problem. You can also try to [deploy your own Vercel instance](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own). I still suspect that the failure of the Rest API is caused by rate limits being hit. When you have your own Vercel instance, the chance of hitting this rate limit is significantly reduced. In the future, you can switch to the [GitHub action](https://github.com/anuraghazra/github-readme-stats/issues/2179).

**Reply by cxhello** [reply]
Ok, thanks a lot for your answer.

**Reply by rickstaa** [reply]
Great. If your issue is fixed, can you maybe `mark as answer` so that it shows up as closed in the discussion tab?

**Reply by cxhello** [reply]
I already mark as answer.

**Reply by rickstaa** [reply]
Ah, I think the new GitHub interface, which I enabled in beta, is broken, as it didn't show up on my side, sorry 😅. Normally there should be a mark as an answer button at the top comment:

![image](https://github.com/anuraghazra/github-readme-stats/assets/17570430/fac9a2c4-ebc2-49ab-b782-bf84695d33c6)

**Reply by rickstaa** [reply]
But on the main discussion page, it now appears as answered, which was what I was after; thanks 👍🏻. 

**Update:** With the new UI disabled, I can select the right answer.

**Reply by cxhello** [reply]
Is there anything else I need to do?

**Reply by rickstaa** [reply]
No, it now works. It was an issue on my side 👍🏻. If you have any more questions, feel free to ask us 🚀.
